src/status_im/contexts/wallet/connected_dapps/view.cljs

```diff
@@ -1,29 +1,128 @@
+                                   :on-disconnect  #(on-disconnect wallet-account dapp)}])}]))
+
+(defn- account-details
+  [{:keys [name emoji color]}]
+  (let [theme (quo.theme/use-theme)]
+    [rn/view {:style style/account-details-wrapper}
+     [quo/context-tag
+      {:theme               theme
+       :type                :account
+       :size                24
+       :account-name        name
+       :emoji               emoji
+       :customization-color color}]]))
+
 (defn- header
-  [{:keys [title subtitle]}]
-  [:<>
-   [rn/view {:style style/header-container}
-    [quo/button
-     {:icon-only?          true
-      :type                :grey
-      :background          :blur
-      :size                32
-      :accessibility-label :close-scan-qr-code
-      :on-press            #(rf/dispatch [:navigate-back])}
-     :i/close]]
-   [quo/text
-    {:size   :heading-1
-     :weight :semi-bold
-     :style  (style/header-text (when subtitle true))}
-    title]])
+  [{:keys [title wallet-account on-close on-add]}]
+  (let [{:keys [color]} wallet-account]
+    [:<>
+     [rn/view {:style style/header-container}
+      [quo/button
+       {:icon-only?          true
+        :type                :grey
+        :background          :blur
+        :size                32
+        :accessibility-label :connected-dapps-close
+        :on-press            on-close}
+       :i/close]]
+     [rn/view {:style style/header-wrapper}
+      [quo/standard-title
+       {:title               title
+        :accessibility-label :connected-dapps
+        :customization-color color
+        :right               [plus-button/plus-button
+                              {:on-press            on-add
+                               :accessibility-label :connected-dapps-add
+                               :customization-color color}]}]
+      [rn/view {:flex 1}
+       [account-details wallet-account]]]]))
 
 (defn view
   []
-  [rn/view {:style {:flex 1}}
-   [header {:title "Connected dApps"}]])
+  (let [{:keys [bottom]}                   (safe-area/get-insets)
+        {:keys [color] :as wallet-account} (rf/sub [:wallet/current-viewing-account])
+        pairings                           (rf/sub [:wallet-connect/pairings])
+        theme                              (quo.theme/use-theme)]
+    [rn/view {:flex 1}
+     [header
+      {:title          (i18n/label :t/connected-dapps)
+       :wallet-account wallet-account
+       :on-close       #(rf/dispatch [:navigate-back])
+       :on-add         #(js/alert "Feature not implemented")}]
+     (if (empty? pairings)
+       [quo/empty-state
+        {:title           (i18n/label :t/no-dapps)
+         :description     (i18n/label :t/no-dapps-description)
+         :image           (resources/get-themed-image :no-dapps theme)
+         :container-style style/empty-container-style}]
+       [rn/view (style/dapps-container bottom)
+        [rn/flat-list
+         {:data                    pairings
```

Comment:
I wonder if we should normalize the pairings data in the db, so that we have in the db a list of pairing topics and a map with topics->pairing-data. This way we would only pass the id to the flatlist item, and the item component can retrieve the pairing data with a subscription. Otherwise, we always have to map over the pairings vector to find the relevant one, which might cause perf issues in the future.

Reply:
But we also want to maintain the order, right? I'd argue that the "display list" operation will be used more often, and, in addition to that, even finding something in a list of 1000 items will take about zero time compared to the amount it will take to redraw the list in the case of update/delete opration anyway.

Reply:
But also I think it doesn't matter. We're here making a list that will contain zero to five items for the majority of users.